src/contexts/AuthContext.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState, useEffect, useContext } from 'react';
+import { auth } from '../utils/firebaseAuthUtil';
+import { useBackend } from './BackendContext';
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  signOut,
+  sendPasswordResetEmail,
+} from 'firebase/auth';
+
+const AuthContext = React.createContext();
+
+export function useAuth() {
```

Comment:
⚠️ **[eslint]** <react-refresh/only-export-components> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components.